src/Nox.Generator/Domain/Entity.template.cs

```diff
@@ -108,6 +109,46 @@ public abstract class {{className}}Base{{ if !entity.IsOwnedEntity }} : {{if ent
     public Nox.Types.{{attribute.Type}}{{if !attribute.IsRequired}}?{{end}} {{attribute.Name}} { get; set; } = null!;
     {{- end}}
 {{- end }}
+{{-if entity.HasDomainEvents}}
+
+	///<inheritdoc/>
+	public IReadOnlyCollection<IDomainEvent> DomainEvents => _domainEvents;
+
+	private readonly List<IDomainEvent> _domainEvents = new();
```

Comment:
can you make this protected, so each service on the partial class can add or override methods and acess the _domainEvents

Reply:
desired changes were applied